### ISSUE TITLE
It could use more of Moo

### ISSUE THREAD

**Issue by dakkar** [perlpunk/App-Spec-p5#14]
Looking at the source code, I notice that in a few places you're not exploiting the facilities that `Moo` provides (e.g. you've got `common` and `build` do more-or-less what `BUILDARGS` and `BUILD` are designed to do, you handle lazy attributes by hand, you have no type constraints).
Is there a particular reason for that, or is just general unfamiliarity with `Moo`?
Would it be appreciated if I sent a pull request making the whole thing more Moo-ish? I'd also like to add some type constraints via `Type::Tiny`, but that can come later.

**Comment by perlpunk** [comment]
yeah, i'm not familiar enough with Moo and did it the old school manual way so far.
I have been refactoring a lot and I hope I have to do less in the future. Would be glad about a PR!
I also wanted to add type constraints someday =)

**Comment by dakkar** [comment]
I'll get started, then ☺
